test/test_sparse_csr.py

```diff
@@ -3998,6 +4000,9 @@ def bsr_dense_linear(input, weights, bias=None):
         def reference(input, mat1, mat2, beta=1, alpha=1):
             assert mat1.layout is torch.strided
             assert mat2.layout is torch.strided
+            if dtype is torch.int8:
+                # workaround RuntimeError: "addmm_cuda" not implemented for 'Char'
+                return beta * input + alpha * (mat1.cpu() @ mat2.cpu()).to(device)
```

Comment:
We have a private api `_int_mm` which seems to support int8 exclusively. Should that be used for the reference computation here? Looks like it depends on `cublasLT` so perhaps it is not available in all builds.

Reply:
If int8 is going to be an important dtype for applications, then I think it would be more beneficial to implement int8 support in addmm and others.
For the reference computations here, basically anything that works across builds will be sufficient for testing purposes.

Reply:
So, the key reason we *have to* use `_int_mm` here is that we need to support `int8@int8 -> int32`. However, `torch.mm` supports `int8@int8->int8`.

We need `int8@int8->int32` both for quantization and also because there's only Tensor core support for this particular output dtype.

@pearu - do you mind trying to add this to `_int_mm` or just add a `_int` variant of the existing sparse BSR Triton kernels that'll be private for now (and that we can use in torchao/superblock)?

Reply:
@cpuhrsch just to be clear, `_int_bsr_dense_linear(input, weights, bias)` ought to return `int32` tensor when all inputs are `int8` tensors? If so, then https://github.com/pytorch/pytorch/pull/133855/files#r1722284345 is wrong, right?

Reply:
Correct. We need it to turn an int32 Tensor.

Reply:
The PR now adds `_int_bsr_dense_linear` to `torch.sparse._triton_ops`.